hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoApproveAllowanceTransitionLogic.java

```diff
@@ -179,42 +177,8 @@ private void removeEntity(final Map<EntityNum, Long> cryptoMap,
 	}
```

Comment:
we can actually remove this method and change to 
```
if (cryptoMap.containsKey(spender.asEntityNum()) && amount == 0) {
 				cryptoMap.remove(spender.asEntityNum());
 			}
 			if (amount > 0) {
 				cryptoMap.put(spender.asEntityNum(), amount);
 				validateAllowanceLimitsOn(accountToApprove, dynamicProperties.maxAllowanceLimitPerAccount());
 			}
 			entitiesChanged.put(accountToApprove.getId().num(), accountToApprove);
```

Reply:
Since we need to change the logic completely , will keep in mind for next PR. Thanks @anighanta